contribs/ev/src/main/java/org/matsim/contrib/ev/strategic/scoring/ChargingPlanScoring.java

```diff
@@ -444,6 +445,12 @@ private void initializeDetours() {
 			AtomicDouble travelDistance = new AtomicDouble(0.0);
 
 			for (Leg leg : TripStructureUtils.getLegs(person.getSelectedPlan())) {
+				// don't we need to filter for legs with the EV, i.e. of the chargingMode?
+				// while the finalizeDetour() method takes max(0,observedDetour) and observedDetour considers trips with EV only,
+				// the fact that we do not filter here means that detours with EV are first compensated by legs with other modes, before they are scored, right?
+				// Moreover, I ran into situations where leg.getTravelTime was undefined, even though that the travel time was set in the route (e.g. walk) -
+				// maybe that doesn't belong here, but should we handle undefined times here?
+				// ts, march 2025
```

Comment:
@sebhoerl : 

I do not really know the reason, but it seems that VSP has scenarios (e.g. Berlin v6.4) where the output plans contain legs for which the travel time is undefined even though that they have a route (with a travel time). 

This is not handled here.

I am currently trying to set up a standalone `SEVC` simulation based on the Berlin output. This is why I stumbled upon that issue.

Reply:
I only realised now that the code here relies on the route being available for the distance, anyway. I will propose to rely on the route for the travel time for a first try and on the leg attribute as a fallback, as well as provide a meaningful error message when both are undefined.

Reply:
I still think we need to filter the leg for `chargingMode` in order to obtain correct detours (haven't checked neither changed the code yet)

Reply:
Yes, I think you're right, we should filter for the chargingMode when initializing the detours

Reply:
For me its fine to get the travel time from the route